tests/end-to-end/external-solution-integration/internal/scenario/scenario.go

```diff
@@ -11,3 +12,38 @@ type Scenario interface {
 	AddFlags(set *pflag.FlagSet)
 	Steps(config *rest.Config) ([]step.Step, error)
 }
+
+type e2eState struct {
```

Comment:
This should be in a separate file together with `comapssE2EState` and `appConnectorE2EState` structs. Maybe `state.go`.